Tests/DistributedActorsTests/RemoteCallTests.swift

```diff
@@ -0,0 +1,330 @@
+        }
+        local.cluster.join(node: remote.cluster.uniqueNode)
+
+        let greeter = Greeter(actorSystem: local)
+        let remoteGreeterRef = try Greeter.resolve(id: greeter.id, using: remote)
+
+        let error = try await shouldThrow {
+            _ = try await remoteGreeterRef.helloThrow(codable: true)
+        }
+        guard error is GenericRemoteCallError else {
+            throw TestError("Expected GenericRemoteCallError, got \(error)")
+        }
+    }
+
+    func test_remoteCall_customCodableErrorAllowList_errorInList() async throws {
+        let local = await setUpNode("local") { settings in
+            settings.remoteCall.codableErrorAllowance = .custom([GreeterCodableError.self, AnotherGreeterCodableError.self])
+            settings.serialization.registerInbound(GreeterCodableError.self)
+            settings.serialization.registerInbound(AnotherGreeterCodableError.self)
+        }
+        let remote = await setUpNode("remote") { settings in
+            settings.remoteCall.codableErrorAllowance = .custom([GreeterCodableError.self, AnotherGreeterCodableError.self])
+            settings.serialization.registerInbound(GreeterCodableError.self)
+            settings.serialization.registerInbound(AnotherGreeterCodableError.self)
+        }
+        local.cluster.join(node: remote.cluster.uniqueNode)
+
+        let greeter = Greeter(actorSystem: local)
+        let remoteGreeterRef = try Greeter.resolve(id: greeter.id, using: remote)
+
+        let error = try await shouldThrow {
+            _ = try await remoteGreeterRef.helloThrow(codable: true)
+        }
+        guard error is GreeterCodableError else {
+            throw TestError("Expected GreeterCodableError, got \(error)")
+        }
+    }
+
+    func test_remoteCall_customCodableErrorAllowList_errorNotInList() async throws {
+        let local = await setUpNode("local") { settings in
+            settings.remoteCall.codableErrorAllowance = .custom([AnotherGreeterCodableError.self])
+            settings.serialization.registerInbound(GreeterCodableError.self)
+            settings.serialization.registerInbound(AnotherGreeterCodableError.self)
+        }
+        let remote = await setUpNode("remote") { settings in
+            settings.remoteCall.codableErrorAllowance = .custom([AnotherGreeterCodableError.self])
+            settings.serialization.registerInbound(GreeterCodableError.self)
+            settings.serialization.registerInbound(AnotherGreeterCodableError.self)
+        }
+        local.cluster.join(node: remote.cluster.uniqueNode)
+
+        let greeter = Greeter(actorSystem: local)
+        let remoteGreeterRef = try Greeter.resolve(id: greeter.id, using: remote)
+
+        let error = try await shouldThrow {
+            _ = try await remoteGreeterRef.helloThrow(codable: true)
+        }
+        guard error is GenericRemoteCallError else {
+            throw TestError("Expected GenericRemoteCallError, got \(error)")
+        }
+    }
```

Comment:
can we add another test when we allow-listed an error, but did not register it?

Reply:
In that case we'd get a timeout error IIRC. What's the expected behavior?

Reply:
We get timeout error because we fail to deserialize `AnyRemoteCallReply` (`TransportPipelines.swift` L641) so we can't even callback.

Reply:
amended https://github.com/apple/swift-distributed-actors/pull/1020/commits/b73914d6a547b2fff6a31fe975363721a7c24982